python-backend/app/api/document_manager/resources/document_manager.py

```diff
@@ -5,110 +5,203 @@
 from datetime import datetime
 from werkzeug.datastructures import FileStorage
 from werkzeug import exceptions
-from flask import request, current_app, send_file
+from flask import request, current_app, send_file, make_response, jsonify
 from flask_restplus import Resource, reqparse
-from flask_uploads import UploadNotAllowed
 
 from ..models.document_manager import DocumentManager
-from app.extensions import api, documents
+from app.extensions import api, cache
 from ...utils.resources_mixins import UserMixin, ErrorMixin
 from ...utils.access_decorators import requires_role_mine_create, requires_any_of, MINE_CREATE, MINESPACE_PROPONENT
 
 
 class DocumentManagerResource(Resource, UserMixin, ErrorMixin):
     parser = reqparse.RequestParser()
-    parser.add_argument('file', type=FileStorage, location='files', action='append')
     parser.add_argument(
         'folder', type=str, required=True, help='The sub folder path to store the document in.')
     parser.add_argument(
         'pretty_folder',
         type=str,
         required=True,
-        help=
-        'The sub folder path to store the document in with the guids replaced for more readable names.'
+        help='The sub folder path to store the document in with the guids replaced for more readable names.'
     )
+    parser.add_argument(
+        'filename', type=str, required=True, help='File name + extension of the document.')
+
+    tus_api_version = '1.0.0'
+    tus_api_supported_versions = '1.0.0'
+    redis_timeout = 86400  # 1 day
```

Comment:
There should be a constant for this in the constants file.

Reply:
Constants have been moved